daffodil-lib/src/main/scala/org/apache/daffodil/lib/util/Misc.scala

```diff
@@ -250,12 +251,9 @@ object Misc {
    * Returns the primary version of daffodil from the jar
    */
   def getDaffodilVersion: String = {
-    val implVersion = this.getClass.getPackage.getImplementationVersion
-    if (implVersion == null) {
-      ""
-    } else {
-      implVersion
-    }
+    val uri = getRequiredResource("org/apache/daffodil/lib/VERSION")
+    val version = Source.fromInputStream(uri.toURL.openStream, "UTF-8").mkString
+    version
```

Comment:
In case it's not obvious to the next reviewer, the managed resource `VERSION` is generated inside the `daffodil-lib/target/classes/org/apache/lib directory` so it doesn't need to be put under source control or excluded from source control.

Reply:
Yep. To be a little bit more exact, the new `genVersion` task writes the version to `daffodil-lib/resource_managed/main/org/apache/daffodil/lib/VERSION`, which sbt copies to `daffodil-lib/target/classes/org/apache/daffodil/lib/` before it generates the jar. 

Both the `resource_managed` and `target` directories are in the .gitignore so the file is never committed, its just generated for every build.